### ISSUE TITLE
manual/javaGuide/main/async/JavaAsync.md

### ISSUE THREAD

**Issue by garbagetown** [playframework-ja/translation-project#550]
- https://github.com/garbagetown/playdocja/blob/2.2.0/documentation/2.2.0/manual/javaGuide/main/async/JavaAsync.md

```
--- /Users/garbagetown/Desktop/2.1.5/manual/javaGuide/main/async/JavaAsync.md   2013-09-19 22:53:02.000000000 +0900
+++ /Users/garbagetown/Desktop/2.2.0/manual/javaGuide/main/async/JavaAsync.md   2013-09-19 10:11:22.000000000 +0900
@@ -2,9 +2,9 @@

 ## Why asynchronous results?

-Until now, we were able to compute the result to send to the web client directly. This is not always the case: the result may depend of an expensive computation or on a long web service call.
+Until now, we have been able to compute the result to send to the web client directly. This is not always the case: the result may depend on an expensive computation or on a long web service call.

-Because of the way Play 2.0 works, action code must be as fast as possible (i.e. non blocking). So what should we return as result if we are not yet able to compute it? The response should be a promise of a result!
+Because of the way Play works, action code must be as fast as possible (i.e. non blocking). So what should we return as result if we are not yet able to compute it? The response should be a promise of a result!

 A `Promise<Result>` will eventually be redeemed with a value of type `Result`. By giving a `Promise<Result>` instead of a normal `Result`, we are able to compute the result quickly without blocking anything. Play will then serve this result as soon as the promise is redeemed. 

@@ -14,60 +14,22 @@

 To create a `Promise<Result>` we need another promise first: the promise that will give us the actual value we need to compute the result:

-'''
-Promise<Double> promiseOfPIValue = computePIAsynchronously();
-Promise<Result> promiseOfResult = promiseOfPIValue.map(
-  new Function<Double,Result>() {
-    public Result apply(Double pi) {
-      return ok("PI value computed: " + pi);
-    } 
-  }
-);
-'''
-
-> **Note:** Writing functional composition in Java is really verbose for the moment, but it should be better when Java supports [[lambda notation| http://mail.openjdk.java.net/pipermail/lambda-dev/2011-September/003936.html]].
-
-Play 2.0 asynchronous API methods give you a `Promise`. This is the case when you are calling an external web service using the `play.libs.WS` API, or if you are using Akka to schedule asynchronous tasks or to communicate with Actors using `play.libs.Akka`.
-
-A simple way to execute a block of code asynchronously and to get a `Promise` is to use the `play.libs.Akka` helpers:
-
-'''
-Promise<Integer> promiseOfInt = Akka.future(
-  new Callable<Integer>() {
-    public Integer call() {
-      return intensiveComputation();
-    }
-  }
-);
-'''
+@[promise-pi](code/javaguide/async/JavaAsync.java)
+
+> **Note:** Writing functional composition in Java is really verbose at the moment, but it should be better when Java supports [lambda notation](http://docs.oracle.com/javase/tutorial/java/javaOO/lambdaexpressions.html).
+
+Play asynchronous API methods give you a `Promise`. This is the case when you are calling an external web service using the `play.libs.WS` API, or if you are using Akka to schedule asynchronous tasks or to communicate with Actors using `play.libs.Akka`.
+
+A simple way to execute a block of code asynchronously and to get a `Promise` is to use the `promise()` helper:
+
+@[promise-async](code/javaguide/async/JavaAsync.java)

 > **Note:** Here, the intensive computation will just be run on another thread. It is also possible to run it remotely on a cluster of backend servers using Akka remote.

 ## AsyncResult

-While we were using `Results.Status` until now, to send an asynchronous result we need an `Results.AsyncResult` that wraps the actual result:
-
-'''java
-public static Result index() {
-  Promise<Integer> promiseOfInt = play.libs.Akka.future(
-    new Callable<Integer>() {
-      public Integer call() {
-        return intensiveComputation();
-      }
-    }
-  );
-  return async(
-    promiseOfInt.map(
-      new Function<Integer,Result>() {
-        public Result apply(Integer i) {
-          return ok("Got result: " + i);
-        } 
-      }
-    )
-  );
-}
-'''
+While we have been simply returning `Result` until now, to send an asynchronous result we need to return `Promise<Result>` from our action:

-> **Note:** `async()` is an helper method building an `AsyncResult` from a `Promise<Result>`.
+@[async](code/javaguide/async/Application.java)

-> **Next:** [[Streaming HTTP responses | JavaStream]]
\ No newline at end of file
+> **Next:** [[Streaming HTTP responses | JavaStream]]

```

**Comment by garbagetown** [comment]
pull requested at #672

**Comment by wm3** [comment]
merged #672
